### ISSUE TITLE
Train a model in parallel?

### ISSUE THREAD

**Issue by manuelreis** [tiny-dnn/tiny-dnn#945]
Hello,

Is there any way of leveraging Tiny DNN for training a model, in parallel?

By parallel, I mean having several threads fetching new training data, computing gradients and updating a set of shared parameters, at the same time.

If possible, how could I achieve that goal?
